Add "profile" loglevel

terminalmage opened the pull request:
This new loglevel sits between INFO and DEBUG, so that logging can be added to
aid in troubleshooting, without having to sift through all the debug logging.

This also adds a decorator which can be used on any function which will log
execution time, and also adds profile logging for template rendering.

terminalmage committed:
Add "profile" loglevel

terminalmage committed:
Add decorator to do profile-level logging for a function

terminalmage committed:
Add profile logging for template rendering

s0undt3ch commented on salt/utils/decorators/__init__.py:
```diff
@@ -140,6 +142,29 @@ class depends(Depends):  # pylint: disable=C0103
+    '''
+    Decorator wrapper to log execution time, for profiling purposes
+    '''
+    def wrapped(*args, **kwargs):
```
You should add a `functions.wraps(function)` decorator to `wrapped` to keep the `wrapped` function attributes equal to those of `function`...

s0undt3ch commented on salt/utils/decorators/__init__.py:
```diff
@@ -1,18 +1,20 @@
 # -*- coding: utf-8 -*-
```
You changed the file perms to 755

terminalmage replied:
That was unintentional, thanks for catching it. I had backed up my git clones to a flash drive when my old work laptop was in the process of dying, and the flash drive was formatted FAT32, so this messed up the file permissions when I copied them back onto the EXT4 partition on my new work laptop.

s0undt3ch commented:
:awesome:

terminalmage committed:
Add @wraps decorator

terminalmage commented:
@s0undt3ch I fixed the perms and rebased that fix into the original commit to ``salt/decorators/__init__.py``. I also added the decorator. Can you have another look? Thanks for the review.

s0undt3ch commented:
:+1:
